lib/services/data/budget/providers/project_budget_task_hierarchy_provider.dart

```diff
@@ -0,0 +1,142 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:seren_ai_flutter/services/data/projects/providers/cur_selected_project_providers.dart';
+import 'package:seren_ai_flutter/services/data/tasks/providers/tasks_by_project_stream_provider.dart';
+import 'package:seren_ai_flutter/services/data/tasks/widgets/gantt/experimental/cur_project_tasks_hierarchy_provider.dart';
+
+final _curProjectBudgetTasksHierarchyProvider =
+    Provider<Map<String, TaskHierarchyInfo>>((ref) {
```

Comment:
can you add high level comments explaining what this is doing? 
for example what is the string -> task hierarchy info what does that represent?

Reply:
so you're making a map of the child/parent relationships of tasks? but why is it called budget? this appears to be task hierarhcy related only? also can you explain why this is needed?